### PR TITLE
Add support for Range requests in file handling

### PR DESCRIPTION
**Pull request by Alirexaa**
This pull request introduces a series of changes to enhance the file handling capabilities of the `chico_server` project, particularly focusing on range requests, error handling, and testing. The most significant changes include adding support for HTTP range requests, improving error handling, and expanding the test coverage.

Enhancements to file handling:

* [`chico_server/src/handlers/file.rs`](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5L93-R180): Added support for HTTP range requests by implementing the `parse_range` function and modifying the `process_file` function to handle range headers. This includes returning appropriate responses for valid and invalid range requests. [[1]](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5L93-R180) [[2]](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5R195-R235)

Error handling improvements:

* [`chico_server/src/handlers/file.rs`](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5L49-L63): Improved error handling in the `FileHandler` by using the `is_ok_and` method to check the existence of the file and returning specific errors for different conditions, such as `ErrorKind::NotFound` and `ErrorKind::IsADirectory`.
* [`chico_server/src/handlers/respond.rs`](diffhunk://#diff-1b6a46b497570178c1f7d01a03b7b2a8672b50fb5b48061bb4677ecc189a721fR72-R76): Added a new method `range_not_satisfiable` to the `RespondHandler` to handle 416 Range Not Satisfiable errors.

Test coverage expansion:

* [`chico_server/src/handlers/file.rs`](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5R314-R390): Added multiple tests to cover the new range request functionality, including tests for valid and invalid ranges, as well as edge cases. [[1]](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5R314-R390) [[2]](diffhunk://#diff-e5a30202ba9014b38734462aea256bebbfd19e1d426ceb3028de12afa071e6e5R598-R729)
* [`chico_server/tests/server.rs`](diffhunk://#diff-85f40e26b2023a99ebcac6a9c742b71d2a6fdbfdbfd1cc08a632f45833c28988R388-R532): Added integration tests to verify the behavior of the file handler for HEAD requests and range requests, ensuring proper headers and responses are returned.

Configuration updates:

* [`chico_server/resources/test_cases/file-handler/file_exist_return_ok.chf`](diffhunk://#diff-9234b42fb3f56812258e1e891a508d8fea4c061cd69a0057514a294e27db6961R6-R9): Added a new route for `test.txt` to facilitate testing of the file handler with different file types.

These changes collectively improve the robustness and functionality of the file handling system in the `chico_server` project, ensuring better compliance with HTTP standards and more comprehensive test coverage.

### CODE REVIEW

File: chico_server/src/handlers/file.rs [hunk 1]
```diff
@@ -76,8 +83,15 @@ impl RequestHandler for FileHandler {
...
         }
+
+        let metadata = tokio::fs::metadata(&path).await;
+        if metadata.is_err() {
+            let err_kind = metadata.as_ref().err().unwrap().kind();
+            return handle_file_error(_request, err_kind).await;
+        }
         let file: File = file.unwrap();
-        process_file(path.to_str().unwrap(), file)
+        let metadata = &metadata.unwrap();
```
[the remark this answers is not in the document]
Comment (Alirexaa) [reply chico_server/src/handlers/file.rs:93]: The given path to `metadata` is different from the first one.
